pep/002-mixin-versioning.md

```diff
@@ -0,0 +1,352 @@
+---
+title: "Mixin Versioning"
+number: "NNN"
+status: "provisional"
+owner: "@carolynvs"
+authors:
+  - "@carolynvs"
+  - "@sestegra"
+created: "2021-02-21"
+type: "feature"
+---
+
+## Abstract
+
+Bundles should be able to specify not only the mixin used, but its version and source.
+Porter should use that information to automatically download the mixins declared by the bundle.
+
+
+## Motivation
+
+Currently bundle authors manage the mixins used within a bundle by hand. 
+They must manually install each mixin using homegrown scripts. 
+Unfortunately the mixin cache is stored in PORTER_HOME, it is impossible to work on bundles that use different versions of the same mixin.
+Porter really needs to manage the mixin cache and appropriately communicating with the correct mixin version targeted by the bundle.
+
+* Mixin authors are impacted by this change because they will need to adjust their mixin declaration schema to allow for declaring the mixin version.
+* Bundle authors are impacted by the new syntax for declaring mixins and no longer managing the bundle cache by hand.
```

Comment:
I'm navigating parsing the impact here and I'm getting a little confused. 

Is it something like Bundle authors are impacted as versioning for mixins will change leading to more usability to be explicit with versioning of mixings but with the new syntax if no version information is specified latest will be used and bundle authors will no longer be able to manage the bundle cache by hand?

Reply:
Right now if I am writing a bundle that needs v1.2.3 of a mixin I have to be aware of which version of that mixin is installed on the computer that is building the bundle. I am seeing authors script out, `porter mixins install MIXIN --version v1.2.3` for each mixin they use, following by a `porter build`. This is the manual cache management that I am referring to. 

With this change, they do not need to write extra scripts around `porter build` and porter will handle that for them. So bundle authors need to start setting a version for each mixin that their bundles use, and stop running extra scripts that they may have written.